File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode2125NumberOfLaserBeamsInABank.cpp
#include "LeetCode2125NumberOfLaserBeamsInABank.h"

int LeetCode2125NumberOfLaserBeamsInABank::numberOfBeams(std::vector<std::string>& bank)
{
    int beamCount{};
    int previousRowDeviceCount{};
    int currentRowDeviceCount{};
    for (auto & row : bank)
    {
        currentRowDeviceCount=0;
        for (auto c:row)
        {
            if (c=='1')
            {
                ++currentRowDeviceCount;
            }
        }
        if (currentRowDeviceCount==0)
            continue;
        beamCount+=currentRowDeviceCount*previousRowDeviceCount;
        previousRowDeviceCount=currentRowDeviceCount;
    }
    return beamCount;
}

File: LeetCodeProblemsGo/LeetCode2125NumberOfLaserBeamsInABank/LeetCode2125NumberOfLaserBeamsInABank.go
package LeetCode2125NumberOfLaserBeamsInABank

func numberOfBeams(bank []string) int {
	beamsCount := 0
	previousRowDevices := 0
	currentRowDevices := 0
	for _, row := range bank {
		currentRowDevices = 0
		for _, char := range row {
			if char == '1' {
				currentRowDevices++
			}
		}
		if currentRowDevices == 0 {
			continue
		}
		beamsCount += previousRowDevices * currentRowDevices
		previousRowDevices = currentRowDevices
	}
	return beamsCount
}
